trulens_eval/trulens_eval/database/sqlalchemy.py

```diff
@@ -694,9 +718,33 @@ def __init__(self):
         self.feedback_columns = set()
 
     def get_df_and_cols(
-        self, apps: Iterable[orm.AppDefinition]
+        self,
+        apps: Optional[List[orm.AppDefinition]] = None,
+        records: Optional[List[orm.Record]] = None
     ) -> Tuple[pd.DataFrame, Sequence[str]]:
-        df = pd.concat(self.extract_apps(apps))
+        """Produces a records dataframe which joins in information from apps and
+        feedback results.
+
+        Args:
+            apps: If given, includes all records of all of the apps in this
+                iterable.
+
+            records: If given, includes only these records. Mutually exclusive
+                with `apps`.
+        """
+
+        assert apps is None or records is None, "`apps` and `records` are mutually exclusive"
```

Comment:
Also regardless, it seems like from L738-L746 it should be at the very least `and` instead of `or`.

Reply:
The apps mode is meant to be independent of the records mode though I see now with the other code changes, the apps mode is never used. Only L601 uses this method and it only gives records due to the change that selects records from the database instead of apps.